kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/ProxyMode.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy;
+
+import java.util.Map;
+import java.util.Optional;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.kroxylicious.proxy.config.Configuration;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * The proxy can operate in one of two modes:
+ * <ol>
+ *     <li>
+ *         {@linkplain #PRODUCTION} mode (default). Development features are disallowed.
+ *     </li>
+ *     <li>
+ *         {@linkplain #DEVELOPMENT} mode. Development features are allowed
+ *     </li>
+ * </ol>
+ */
+public enum ProxyMode {
+    PRODUCTION {
+        @Override
+        public Configuration validate(@NonNull Configuration config) {
+            Optional<Map<String, Object>> development = config.development();
+            if (development.isPresent() && !development.get().isEmpty()) {
+                LOG.error("development configuration for proxy present, but development mode not enabled: {}", development.get());
+                throw new IllegalConfigurationException("development configuration for proxy present, but development mode not enabled");
```

Comment:
I wonder if we should reduce this to a `warn` and just have the runtime operate in production mode and *ignore* the development configuration. That feels like a more flexible model which would allow people to promote a config file between environments. 

E.g. run DEV/Integration environments in `DEVELOPMENT` mode for some additional validation and then promote that configuration to UAT/Stage/Prod in `PRODUCTION` mode.